src/index.ts

```diff
@@ -336,49 +340,34 @@ class Logging {
   createSink(
       name: string, config: CreateSinkRequest,
       callback: CreateSinkCallback): void;
-  createSink(
-      name: string, config: CreateSinkRequest,
-      callback?: CreateSinkCallback): Promise<[Sink, LogSink]>|void {
-    const self = this;
-    if (!is.string(name)) {
+  async createSink(name: string, config: CreateSinkRequest):
+      Promise<[Sink, LogSink]> {
+    if (typeof name !== 'string') {
       throw new Error('A sink name must be provided.');
     }
-    if (!is.object(config)) {
+    if (typeof config !== 'object') {
       throw new Error('A sink configuration object must be provided.');
     }
     if (common.util.isCustomType(config.destination, 'bigquery/dataset')) {
-      this.setAclForDataset_(name, config, callback!);
-      return;
+      await this.setAclForDataset_(config);
     }
     if (common.util.isCustomType(config.destination, 'pubsub/topic')) {
-      this.setAclForTopic_(name, config, callback!);
-      return;
+      await this.setAclForTopic_(config);
     }
     if (common.util.isCustomType(config.destination, 'storage/bucket')) {
-      this.setAclForBucket_(name, config, callback!);
-      return;
+      await this.setAclForBucket_(config);
     }
     const reqOpts = {
       parent: 'projects/' + this.projectId,
       sink: extend({}, config, {name}),
     };
     delete reqOpts.sink.gaxOptions;
-    this.request(
-        {
-          client: 'ConfigServiceV2Client',
-          method: 'createSink',
-          reqOpts,
-          gaxOpts: config.gaxOptions,
-        },
-        (err, resp) => {
-          if (err) {
-            callback!(err, null, resp);
-            return;
-          }
-          const sink = self.sink(resp.name);
-          sink.metadata = resp;
-          callback!(null, sink, resp);
-        });
+    await this.setProjectId(reqOpts);
```

Comment:
It's awesome seeing this pattern, where we can get the project ID in an un-obtrusive way with await 👍 I think we have a chance to ditch the `{{projectId}}` placeholder token here altogether. If we just get the project ID before we build the `reqOpts` object (L#360), then we know `this.projectId` is the right value-- no need to use the token for later replacement, since if we can't get it now, we can't get it later, either.

Reply:
Tagging related issue which seeks to destroy the placeholder: https://github.com/googleapis/nodejs-common/issues/10

Reply:
This method is dependent on objects from other libraries `storage.bucket`, `bigQuery.dataset` or `pubsub.topic` which could still have `{{projectId}}` placeholder. I would have to still run `replaceProjectIdToken` after building the `reqOpts` object. So I placed it after, to have it in one place for both.

Reply:
Good catch, that is reasonable. I believe this specific method is unique, so I might have picked a bad one to make this suggestion on. For the others, could we skip the `setProjectId()` call, and instead just put the real projectId right where we need it in the reqOpts?

Reply:
@stephenplusplus 
It looks like I can skip `setProjectId()` `getSinks` and `getSinksStream` since these methods do not depend on the `Sink` constructor.

I am leaving `getEntries` and `getEntriesStream` as is since these guys do depend on the `Log` constructor and I would still need to to `replaceProjectIdToken` for `options.filter` 

https://github.com/googleapis/nodejs-logging/blob/81138d6a5a302b46a8a8bb87a58c320dff1574d0/src/log.ts#L525-L530
from 
https://github.com/googleapis/nodejs-logging/blob/81138d6a5a302b46a8a8bb87a58c320dff1574d0/src/log.ts#L108-L110

Reply:
For `getEntries`/`getEntriesStream`, I think we can do the refresh right there:

```ts
this.projectId = await this.auth.getProjectId()
this.formattedName_ = Log.formatName(this.projectId, this.name)
options = extend({ filter: `logName="${this.formattedName_}"` })

// for getEntriesStream
this.auth.getProjectId().then(projectId => {
  this.projectId = projectId
  this.formattedName_ = Log.formatName(this.projectId, this.name)
  options = extend({ filter: `logName="${this.formattedName_}"` })
})
```

The mission for a while has been to eliminate that token altogether, but it was the absence of async/await that has prevented it. But now that we have it, anywhere `this.formattedName_` is used, it should be replaced with code similar to the above, in this file and others where possible. Just a thought, but maybe `formatName()` should be repurposed to do the project ID fetching as well.